includes/init_includes/init_https.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/** 
+ * Do automatic redirect of http URIs to https, if https is given in HTTP_SERVER in configure.php
+ *
+ */
+
+if ((IS_ADMIN_FLAG || HTTPS_SERVER == HTTP_SERVER) && (strpos(HTTP_SERVER, 'https:') !== false)) {
```

Comment:
I've been trying to find a way to express the below based on previous conversations. I've spent too much time worrying about that so here's my thoughts on the structure of the check(s) to lead to the redirect action.

Unless there is a specific reason to have the `$request_type` check occur after all of the other checks (as provided which for a site that is being accessed via https would be performed on each and every page load), then would suggest moving the last `if` up to the beginning of the first `if`.  Then, with regards to the `HTTPS_SERVER` to `HTTP_SERVER` comparison, this seems like it is only important if `ENABLE_SSL` is set to `true` otherwise in theory nothing else would need to access/use `HTTPS_SERVER` and related constants.  Then would also need to check for the condition where `ENABLE_SSL != 'true'` and if all of those things are `true` by this point, then seems should be checking for https: to begin the `HTTP_SERVER` constant, not just be somewhere in the string. (could have an internal server configuration that places some variation of https at the end of the string and use a port such that the string would end with '...https:XXX' but yet begin with http:).  All this making the above line:
````diff
- if ((IS_ADMIN_FLAG || HTTPS_SERVER == HTTP_SERVER) && strpos(HTTP_SERVER, 'https:') !== false) {
+ if ($request_type !== 'SSL' && (IS_ADMIN_FLAG || ENABLE_SSL == 'true' && HTTPS_SERVER == HTTP_SERVER || ENABLE_SSL != 'true') && strpos(HTTP_SERVER, 'https:') === 0) {
`````

Reply:
I think I understand your thinking here, and agree with the suggestion to rearrange the order of things. 

But I wonder whether the check for `ENABLE_SSL` is necessary.  
My personal preference is to dump that define altogether: the thinking being that since the google gods are driving the web industry to use https everywhere, then there's no need to even use ENABLE_SSL at all.  And probably no need for HTTPS_SERVER either.  A single define of HTTP_SERVER would suffice for the entire site. There'd be no need for switching between http/https either. 
I don't see the need for supporting these extra defines in either production or dev environments. 

Apart from backward-compatibility, is there a technical or business reason to keep them at all?

Reply:
The only thing that seems to come to mind and likely should be something independently addressed outside of the core of ZC would be some sort of "alternate" server address from which "non-https" content (under the guise of either using the alternate define of what the main store uses and then using or not using https) could be pulled/obtained. I have no such need nor have I worked with anyone that has shown such a need. Other plugins have offered "remote" server related code for those that say split their images off to another server.

To me it would seem that if the thought is all https: or all not as appears to be the direction of the pendulum, that there is no future need for a second define nor an indicator of the ability to switch and from that a need to switch to/from the same server but via a different "mode" of communication.  (ie. be gone HTTPS_SERVER and ENABLE_SSL) I don't know what forum of nor group of people need to be canvassed to elicit more of a technical/business need response.  

Historical conversations were that encrypting added a system burden (burden = slowdown = reduced rankings) and that where there wasn't a need for some form of increased data protection then don't use https. That's perhaps the old mantra and ZC has been handling switching between the two states well, but as pointed out and as the discussions over the last couple of years have gone, the way suggested to go is all one way all of the time with "preference" of https. 

Makes me wonder if it is worth digging into all of the code to remove the flips and flops or would a few key extractions address the issue and allow someone to revive it if the need resurfaced? Ie. bite the bullet or hoard?